src/Compilers/CSharp/Portable/Symbols/SymbolWithAnnotations.cs

```diff
@@ -615,14 +690,17 @@ private sealed class LazyNullableTypeParameter : TypeSymbolWithAnnotations
             public LazyNullableTypeParameter(CSharpCompilation compilation, TypeSymbolWithAnnotations underlying)
             {
                 Debug.Assert(compilation.IsFeatureEnabled(MessageID.IDS_FeatureStaticNullChecking));
-                Debug.Assert(underlying.IsNullable != true);
+                Debug.Assert(!underlying.IsAnnotated);
                 Debug.Assert(underlying.TypeKind == TypeKind.TypeParameter);
                 Debug.Assert(underlying.CustomModifiers.IsEmpty);
                 _compilation = compilation;
                 _underlying = underlying;
             }
 
             public override bool? IsNullable => true;
+            public override bool IsAnnotated => true;
+            protected override AnnotationKind Annotation => AnnotationKind.Annotated;
+            public override bool NonNullTypes => _underlying.NonNullTypes;
```

Comment:
Since `NonNullTypes` is irrelevant when `IsAnnotated` is true, maybe we should just return a fixed value here? It would be even better if we could throw, but I don't think that's possible at the moment (there are some unconditional callers).